ledger/src/executor.rs

```diff
@@ -0,0 +1,421 @@
+use {
+    crate::{
+        blockstore_processor::{BlockCostCapacityMeter, TransactionStatusSender},
+        token_balances::collect_token_balances,
+    },
+    crossbeam_channel::{unbounded, Receiver, RecvError, SendError, Sender},
+    rayon::{
+        iter::{IndexedParallelIterator, IntoParallelRefIterator, ParallelIterator},
+        ThreadPool,
+    },
+    solana_program_runtime::timings::ExecuteTimings,
+    solana_rayon_threadlimit::get_max_thread_count,
+    solana_runtime::{
+        bank::{Bank, TransactionResults},
+        bank_utils,
+        transaction_batch::TransactionBatch,
+        vote_sender_types::ReplayVoteSender,
+    },
+    solana_sdk::{
+        clock::MAX_PROCESSING_AGE,
+        feature_set,
+        pubkey::Pubkey,
+        signature::Signature,
+        transaction::{Result, SanitizedTransaction, TransactionAccountLocks, TransactionError},
+    },
+    solana_transaction_status::token_balances::TransactionTokenBalancesSet,
+    std::{
+        borrow::Cow,
+        collections::{HashMap, HashSet},
+        result,
+        sync::{Arc, RwLock},
+        thread::{self, Builder, JoinHandle},
+    },
+};
+
+pub(crate) struct TransactionBatchWithIndexes<'a, 'b> {
+    pub batch: TransactionBatch<'a, 'b>,
+    pub transaction_indexes: Vec<usize>,
+}
+
+/// Callback for accessing bank state while processing the blockstore
+pub type ProcessCallback = Arc<dyn Fn(&Bank) + Sync + Send>;
+
+// get_max_thread_count to match number of threads in the old code.
+// see: https://github.com/solana-labs/solana/pull/24853
+lazy_static! {
+    static ref PAR_THREAD_POOL: ThreadPool = rayon::ThreadPoolBuilder::new()
+        .num_threads(get_max_thread_count())
+        .thread_name(|ix| format!("solBstoreProc{:02}", ix))
+        .build()
+        .unwrap();
+}
+
+const DEFAULT_CONFLICT_SET_SIZE: usize = 30;
+
+pub struct ReplayResponse {
+    pub result: Result<()>,
+    pub timing: ExecuteTimings,
+    pub batch_idx: Option<usize>,
+}
+
+/// Request for replay, sends responses back on this channel
+pub struct ReplayRequest {
```

Comment:
> Actually, now that I've looked further down it seems the executor (Replayer) is more generic and can have multiple _unrelated_ threads sending tx to be processed. So I'm not sure the above comment is valid, I'm going to leave it though.

yeah, i noticed that we replay active banks in parallel now, which means you might have muliple banks being scheduled here. 

my 1.10 PR had the same response sender/receiver for the singleton replayer, but that breaks down when you have multiple banks being replayed at the same time. instead of dynamically creating and destroying replayer, figure each replay handle can have its own response sender/receiver combo which lets us create a different replayer handle for each bank being replayed. that's probably cheaper than spinning up/tearing down threads when we hit the parallel replay condition